### PR TITLE
support ad-hoc polygons in query

### PR DESCRIPTION
**Pull request by shz117**
- support ad-hoc polygon in a new filter `in_polygon(<point_column>, <polygon in WKT format>)`
 - allow exporting polygon as virtual dimension "_geoshape"

### CODE REVIEW

File: query/aql_compiler.go [hunk 1]
```diff
@@ -967,6 +975,42 @@ func (qc *AQLQueryContext) Rewrite(expression expr.Expr) expr.Expr {
...
 
 			e.ExprType = expr.Boolean
+		case expr.PointInPolygonCallName:
+			if len(e.Args) != 2 {
+				qc.Error = utils.StackError(
+					nil, "expect 2 arguments for %s, but got %s", e.Name, e.String())
+				break
+			}
+			lhsRef, isVarRef := e.Args[0].(*expr.VarRef)
+			if !isVarRef || (lhsRef.DataType != memCom.GeoShape && lhsRef.DataType != memCom.GeoPoint) {
```
Comment (jshencode) [inline query/aql_compiler.go]: lhs can be GeoShape?

File: query/aql_compiler.go [hunk 2]
```diff
@@ -967,6 +975,42 @@ func (qc *AQLQueryContext) Rewrite(expression expr.Expr) expr.Expr {
 			}
 
 			e.ExprType = expr.Boolean
+		case expr.PointInPolygonCallName:
```
Comment (jshencode) [inline query/aql_compiler.go]: can we still use geo_intersect call for this case

Comment (shz117) [reply query/aql_compiler.go]: it's going to be tricking to reuse the name. as in qc.rewrite, we validate arg types base on call name, but not where the expr came from.

File: query/aql_compiler.go (hunk 1 above)
Comment (shz117) [reply query/aql_compiler.go]: nope. good catch.

**Comment by shz117** [comment]
close for now, might need major refactor

**Pull request closed without merging**